Assets/ScriptableObjects/PlayerResource.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using System;
+
+[CreateAssetMenu(fileName = "New Player Resource", menuName = "PlayerResource")]
+
+public class PlayerResource : ScriptableObject
+{
+    [SerializeField] new string name;
+    [SerializeField] int maxValue;
+    [SerializeField] int minValue;
+    [SerializeField] int canvasChildNumber;
+    [SerializeField] string staticText;
+
+    private int currentValue;
+    private Canvas canvas;
+
+    private void OnEnable()
```

Comment:
This lifecycle method differs from a Monobehaviour - it is true that `OnEnable` will be fired when this object is first loaded, I'm guessing when a MonoBehaviour loads and holds a reference to it. [Documentation isn't super clear here](https://docs.unity3d.com/ScriptReference/ScriptableObject.html). Definitely worth testing.

Reply:
Recommend changing this method name to `Init` and calling it in the `Awake` method of the MonoBehaviour that references it. This way, it's more explicit. However, `OnEnable` might also work here.